tests/test_manylinux.py

```diff
@@ -225,6 +225,46 @@ def test_build_wheel_with_binary_executable(docker_container):
         python_id, ['python', '-c', 'from testpackage import runit; print(runit(1.5))']).strip()
     assert output.strip() == '2.25'
 
+@pytest.mark.parametrize('with_dependency', ['0', '1'])
+def test_build_wheel_with_highest_policy_and_image_dependencies(with_dependency, docker_container):
+    # Test building a wheel that contains shared library dependencies
+    # The python extension module itself only require manylinux1, its dependencies shall require the highest level possible
+
+    policy, manylinux_id, python_id, io_folder = docker_container
+
+    docker_exec(manylinux_id, ['bash', '-c', 'cd /auditwheel_src/tests/testdependencies && WITH_DEPENDENCY={} python setup.py -v build_ext -f bdist_wheel -d /io'.format(with_dependency)])
+
+    filenames = os.listdir(io_folder)
+    orig_wheel = filenames[0]
+    assert 'manylinux' not in orig_wheel
+
+    # Repair the wheel using the appropriate manylinux container
+    repair_command = (
+        'auditwheel -v repair --plat manylinux1_x86_64 -w /io /io/{orig_wheel}'
+    ).format(policy=policy, orig_wheel=orig_wheel)
+    try:
+        docker_exec(manylinux_id, ['bash', '-c', 'LD_LIBRARY_PATH=/auditwheel_src/tests/testdependencies ' + repair_command])
+        assert policy == 'manylinux1'
+    except CalledProcessError:
+        assert policy != 'manylinux1'
```

Comment:
I follow how the test works here, but the logic is kind of inside out. We're really testing that it succeeds when built on the manylinux1 image but not when built on the manylinux2010 image - because then it uses a newer symbol and isn't manylinux1 compatible.

But what we actually assert is that if repairing works, then it must be a manylinux1 image in use, and if it fails, it must *not* be. The dependent and independent variables are swapped around.

That's OK, and I can't immediately see a neat way to write the test without doing that. But let's have an explanatory comment, and maybe a helpful message if the assertions fail.

Reply:
> But let's have an explanatory comment

I realize now that the comments does not reflect what's done...
I will have a hard time explaining what's done with decent english but let's try. If you have a better suggestion, please do tell. If you think that's good enough, I'll update the commit.
```
# Always try to repair the wheel targeting manylinux1 policy

# with_dependency == 0
# The python module has no dependencies that should be grafted-in and
# uses versioned symbols not available on policies pre-dating the policy matching the image being tested.

# with_dependency == 1
# The python module itself does not use versioned symbols but has a dependency that should be grafted-in that
# uses versioned symbols not available on policies pre-dating the policy matching the image being tested.

# Since manylinux1 has no pre-dating policy, it's the only image/policy that should be able to repair those wheels.
```

> The dependent and independent variables are swapped around.

Maybe something like that would work:
```
repair_command = (
	        'LD_LIBRARY_PATH=/auditwheel_src/tests/testdependencies auditwheel -v repair --plat manylinux1_x86_64 -w /io /io/{orig_wheel}'
	    ).format(policy=policy, orig_wheel=orig_wheel)
if policy != 'manylinux1':
    with self.assertRaises(CalledProcessError, msg='Repairing this wheel targeting manylinux1  shall fail on {policy} image'.format(policy=policy)):
        docker_exec(manylinux_id, ['bash', '-c', repair_command])
    return

# go-on checking all works properly on manylinux1 image
docker_exec(manylinux_id, ['bash', '-c', repair_command])
...
```

Reply:
I think the rearrangement you propose is an improvement - it's more verbose, but it makes the intent more clear.

I'm happy to have a go at clarifying any comments I can understand, as I am a native English speaker. :slightly_smiling_face:

Reply:
Should be better now. I force-pushed a new commit for tests.